File: Gestion A C/models/Vente.php
<?php
require_once 'Operation.php';
require_once 'ArticleVente.php';
require_once 'Client.php';
require_once 'Sys.php';

class Vente extends Operation {
    private static int $nbV = 0;
    private Client $client;

    public function __construct() {
        parent::__construct();
        self::$nbV++;
        $this->article = new ArticleVente();
        $this->id = "V-" . self::$nbV;
        $this->nature = "Vente";
    }

    public static function getNbV(): int {
        return self::$nbV;
    }

    public static function setNbV(int $nbV): void {
        self::$nbV = $nbV;
    }

    public function getClient(): Client {
        return $this->client;
    }

    public function setClient(Client $client): void {
        $this->client = $client;
    }

    public function __toString(): string {
        return "\n---------------------------------------\n" .
            "ID                  : " . $this->id . "\n" .
            "ID du client        : " . $this->client->getId() . "\n" .
            "Telephone du client : " . $this->client->getTelephone() . "\n" .
            "Nom Article         : " . Sys::capitalize($this->article->getNomArticle()) . "\n" .
            "Prix                : " . $this->article->getPrix() . " Fcfa\n" .
            "Quantite            : " . $this->article->getQuantite() . "\n" .
            "Montant             : " . $this->article->getMontant() . " Fcfa\n" .
            "Nature              : " . $this->nature . "\n" .
            "Date                : " . $this->date;
    }
}

File: Gestion Atelier De Couture/src/entity/Vente.java
package entity;

import services.Sys;

public class Vente extends Operation {

    private static int nbV;
    private Client client;

    public Vente() {
        super();
        nbV++;
        this.article = new ArticleVente();
        this.id = "V-" + nbV;
        this.nature = "Vente";
    }

    public static int getNbV() {
        return nbV;
    }

    public static void setNbV(int nbV) {
        Vente.nbV = nbV;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }

    public String toString() {
        return "\n---------------------------------------\n" +
                "ID                  : " + id + "\n" +
                "ID du client        : " + client.getId() + "\n" +
                "Telephone du client : " + client.getTelephone() + "\n" +
                "Nom Article         : " + Sys.capitalize(article.getNomArticle()) + "\n" +
                "Prix                : " + article.getPrix() + " Fcfa" + "\n" +
                "Quantite            : " + article.getQuantite() + "\n" +
                "Montant             : " + article.getMontant() + " Fcfa" + "\n" +
                "Nature              : " + nature + "\n"+
                "Date                : " + date;
    }

}
